discovery-provider/src/challenges/challenge_event_bus.py

```diff
@@ -55,6 +55,32 @@ def dispatch(
         except Exception as e:
             logger.warning(f"ChallengeEventBus: error enqueuing to Redis: {e}")
 
+    def queue_event(
```

Comment:
I think the API of this class is getting a bit confusing now, given that it has both a dispatch and enqueue method, and it's not clear which should be publicly called. I think a cleaner way might be only have dispatch (or enqueue) be public, but inside dispatch, just dispatch to your internal queue, which is then flushed to Redis with your `flush_event_queue` method.

Reply:
Updated to use a nice `with` style after some experiments - let me know what you think